djautotask/api_rest.py

```diff
@@ -291,21 +291,33 @@ def fetch_resource(self, next_url=None, retry_counter=None, *args,
                wait_exponential_multiplier=RETRY_WAIT_EXPONENTIAL_MULTAPPLIER,
                wait_exponential_max=RETRY_WAIT_EXPONENTIAL_MAX,
                retry_on_exception=retry_if_api_error)
-        def _fetch_resource(endpoint, retry_counter=None, **kwargs):
-            if not retry_counter:
-                retry_counter = {'count': 0}
-            retry_counter['count'] += 1
+        def _fetch_resource(endpoint_url, request_retry_counter=None,
+                            request_method=method, request_body=body,
+                            **kwargs):
+            if not request_retry_counter:
+                request_retry_counter = {'count': 0}
+            request_retry_counter['count'] += 1
 
             try:
-                logger.debug('Making GET request to {}'.format(endpoint))
-                response = requests.get(
-                    endpoint,
+                logger_message = 'Making {} request to {}'.format(
+                    request_method.upper(), endpoint)
+                if body:
+                    logger_message = \
+                        '{}. Request body: {}'.format(logger_message, body)
+
+                logger.debug(logger_message)
```

Comment:
1. if body:
    Do we need body here? I think in the case of POST, we can add log to show the contents of POST body. In this case, the code can be simplified in the future:

```
        self.build_query_string(**kwargs)

        return self.fetch_resource(next_url, method='post',
                                   body=self.QUERYSTR, *args, **kwargs)
```
to
```
        return self.fetch_resource(next_url, method='post', *args, **kwargs)
```
2. logger_message in line 305 override line 302. Do you intend to concatenate logs here ?

Reply:
1. My intent was to have a debug log to show the POST conditions since when using GET you can see the conditions in the URL. If you can improve the code and still ensure we can see debug logs showing query conditions for both GET and POST let me know.

I don't understand how we can simplify:
```python
        self.build_query_string(**kwargs)

        return self.fetch_resource(next_url, method='post',
                                   body=self.QUERYSTR, *args, **kwargs)
```
to 
```python
        return self.fetch_resource(next_url, method='post', *args, **kwargs)
```
can you explain that a bit more?

Reply:
2. Yes I want to concat those logs. Maybe I'm missing something - I thought line 305 would take the existing logger_message and append the request body conditions/additional text.

Reply:
```
self.build_query_string(**kwargs)
body=self.QUERYSTR
```
This can be done inside of `self.fetch_resource` when self.QUERYSTR is None. So, we can just say:
`self.fetch_resource(next_url, method='post', *args, **kwargs)`.

I thought you need to pass body parameter to display POST body for logging. If this is the only purpose, I think we can just say:

`if body:` to `if request_method.upper() == 'POST':` or something like this.

Do I miss something important? Then, please let me know.

Reply:
Thanks for clarifying. I think I get it now and yeah that does simplify things. I'll work on making those changes.

Reply:
Nice suggestions. I think it's cleaner now.

Reply:
Just added another fix cause I realized I was send QUERYSTR as the body even on GET requests. Should be fixed now.